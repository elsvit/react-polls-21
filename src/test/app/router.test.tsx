import * as React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Router from '~/AppRouter';
import {createMemoryHistory} from 'history';

describe('Router', () => {
  it('should render without error', () => {
    const Renderer = ShallowRenderer.createRenderer();
    const history = createMemoryHistory();

    Renderer.render(<Router history={history} />);

    const wrapper = Renderer.getRenderOutput();

    expect(wrapper).toMatchSnapshot();
  });
});