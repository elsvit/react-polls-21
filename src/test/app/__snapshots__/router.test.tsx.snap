// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Router should render without error 1`] = `
<Memo(Connect(ConnectedRouterWithContext))
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <DrawerWrapper>
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/questions/:id"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "type": [Function],
          }
        }
        exact={true}
        path="/questions"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/add"
      />
      <Redirect
        to="/questions"
      />
    </Switch>
  </DrawerWrapper>
  <NotificationContainer
    enterTimeout={400}
    leaveTimeout={400}
  />
</Memo(Connect(ConnectedRouterWithContext))>
`;
